### PR TITLE
feat(files): add a tab on the person page to create, list, download a…

### PR DESCRIPTION
**Pull request by jnizet**
…nd delete files (documents) for that person

The files are stored in Google Cloud Storage. See the README for details.

This is a WIP: tests are missing. But the feature is usable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,13 +77,21 @@ This will build a standalone jar at `backend/build/libs/globe42.jar`, that you c
...
-
 To start the application with the demo profile, add this command-line option:  
     
     --spring.profiles.active=demo
     
 And the full app runs on http://localhost:9000
 
+By default, the default GCS credentials are used when launching the app this way. That means
+that the GCS APIs won't be accessible unless you set the `GOOGLE_APPLICATION_CREDENTIALS` as
+described in [the documentation about default credentials](https://developers.google.com/identity/protocols/application-default-credentials#howtheywork]
```
Comment (acrepet) [inline README.md]: last parenthesis is missing : (https://developers.google.com/identity/protocols/application-default-credentials#howtheywork]

File: README.md [hunk 2]
```diff
@@ -41,7 +41,32 @@ Then run `yarn start` to start the app, using the proxy conf to reroute calls to
...
+Two separate accounts, and thus credentials files, are used:
+
+ - one for development
+ - one for production
+ 
+We expect to stay inside the "Always Free" tier, whatever the account is.
+
+Here are various interesting links regarding Google Cloud Storage (GCS):
+
+ - [https://cloud.google.com/storage/](The landing page of GCS)
```
Comment (acrepet) [inline README.md]: reverse () and [] : [https://cloud.google.com/storage/](The landing page of GCS) -> [The landing page of GCS](https://cloud.google.com/storage/)

File: README.md [hunk 3]
```diff
@@ -41,7 +41,32 @@ Then run `yarn start` to start the app, using the proxy conf to reroute calls to
...
+ - [https://cloud.google.com/storage/pricing](The pricing page of GCS). It describes the conditions
```
Comment (acrepet) [inline README.md]: reverse it : [https://cloud.google.com/storage/pricing](The pricing page of GCS) -> [The pricing page of GCS](https://cloud.google.com/storage/pricing)

File: README.md [hunk 4]
```diff
@@ -41,7 +41,32 @@ Then run `yarn start` to start the app, using the proxy conf to reroute calls to
...
+
+Here are various interesting links regarding Google Cloud Storage (GCS):
+
+ - [https://cloud.google.com/storage/](The landing page of GCS)
+ - [https://cloud.google.com/storage/pricing](The pricing page of GCS). It describes the conditions 
+   of the "Always Free" tier. The most importtant part being that (as of now), the only regions 
+   eligible to this free tier are `us-west1`, `us-central1`, and `us-east1`. This means that the buckets
+   created for the application should be created in one of these 3 regions. We currently use only one bucket 
+   (for each account - dev and prod)
+ - [https://console.cloud.google.com/storage](The Google Cloud console): Make sure to select the appropriate
```
Comment (acrepet) [inline README.md]: reverse it [The Google Cloud console](https://console.cloud.google.com/storage)

File: README.md [hunk 5]
```diff
@@ -41,7 +41,32 @@ Then run `yarn start` to start the app, using the proxy conf to reroute calls to
...
+ - [https://cloud.google.com/storage/](The landing page of GCS)
+ - [https://cloud.google.com/storage/pricing](The pricing page of GCS). It describes the conditions 
+   of the "Always Free" tier. The most importtant part being that (as of now), the only regions 
+   eligible to this free tier are `us-west1`, `us-central1`, and `us-east1`. This means that the buckets
+   created for the application should be created in one of these 3 regions. We currently use only one bucket 
+   (for each account - dev and prod)
+ - [https://console.cloud.google.com/storage](The Google Cloud console): Make sure to select the appropriate 
+   Google account when visiting it, and to select the `globe42` project. It allows creating the bucket,
+   browsing and deleting the files, creating credentials (service accounts), etc.
+ - [https://cloud.google.com/storage/docs/](The documentation), and
```
Comment (acrepet) [inline README.md]: reverse it !

Comment (acrepet) [inline README.md]: reverse it!

File: README.md [hunk 6]
```diff
@@ -41,7 +41,32 @@ Then run `yarn start` to start the app, using the proxy conf to reroute calls to
...
+ - [https://cloud.google.com/storage/pricing](The pricing page of GCS). It describes the conditions 
+   of the "Always Free" tier. The most importtant part being that (as of now), the only regions 
+   eligible to this free tier are `us-west1`, `us-central1`, and `us-east1`. This means that the buckets
+   created for the application should be created in one of these 3 regions. We currently use only one bucket 
+   (for each account - dev and prod)
+ - [https://console.cloud.google.com/storage](The Google Cloud console): Make sure to select the appropriate 
+   Google account when visiting it, and to select the `globe42` project. It allows creating the bucket,
+   browsing and deleting the files, creating credentials (service accounts), etc.
+ - [https://cloud.google.com/storage/docs/](The documentation), and 
+   [the javadoc of the Java client library](https://googlecloudplatform.github.io/google-cloud-java/0.23.1/apidocs/index.html)
```
Comment (acrepet) [inline README.md]: reverse it!

**Comment by jnizet** [comment]
@acrepet I fixed the issues in the README.

Regarding the name/label associated with a file. This could be done without too much problem, since both GCS and S3 allow for metadata associated with a file. But I wanted to keep is straight simple for a first iteration. I have the feeling that forcing them to choose good, meaningful file names in the first place would be a good idea, and make things simpler and safer for everyone.

Regarding the access to the GCS console for the account used in dev, you just use your Ninja Squad Google account (agnes@ninja-squad.com) and select the globe42 project. For prod, Globe should create their own account and their own google cloud project.

**Comment by acrepet** [comment]
Regarding the name/label associated with a file : OK for keeping this feature straight simple for a first iteration!
And tanks for having explain me the access to the GCS console

**Pull request merged**